include/execution.hpp

```diff
@@ -117,22 +117,61 @@ namespace std::execution {
       receiver_of<_Receiver, _As...> &&
       nothrow_tag_invocable<set_value_t, _Receiver, _As...>;
 
+  /////////////////////////////////////////////////////////////////////////////
+  // receiver_signatures
+  // NOT TO SPEC
+  namespace __receiver_signatures {
+    template <same_as<set_value_t> _Tag, class _Ty = __q<__types>, class... _Args>
+      __types<__minvoke<_Ty, _Args...>> __test(_Tag(*)(_Args...));
+    template <same_as<set_error_t> _Tag, class _Ty = __q<__types>, class _Error>
+      __types<__minvoke1<_Ty, _Error>> __test(_Tag(*)(_Error));
+    template <same_as<set_done_t> _Tag, class _Ty = __q<__types>>
+      __types<__minvoke<_Ty>> __test(_Tag(*)());
+    template <class, class = void>
+      __types<> __test(...);
+
+    template <class _Sig, class _Tag, class _Ty = __q<__types>>
+      using __signal_args_t =
+        decltype(__test<_Tag, _Ty>((_Sig*) nullptr));
+
+    template <class _Sig>
+      concept __receiver_signal =
+        requires { typename __id<decltype(__test((_Sig*) nullptr))>; };
```

Comment:
Won't this check also match the ellipsis-taking overload of `__test()` above?
Is this supposed to just be testing that the signature is one of the set_value/set_error/set_done signatures?

Reply:
No, because the non-defaulted template parameter is non-deducible.